lib/logstash/inputs/s3.rb

```diff
@@ -287,26 +305,40 @@ def process_log(queue, key)
     object = @s3bucket.objects[key]
 
     filename = File.join(temporary_directory, File.basename(key))
-
-    download_remote_file(object, filename)
-
-    process_local_log(queue, filename)
-
-    backup_to_bucket(object, key)
-    backup_to_dir(filename)
-
-    delete_file_from_bucket(object)
-    FileUtils.remove_entry_secure(filename, true)
+    
+    if download_remote_file(object, filename)
```

Comment:
Yes this was needed.

You can download pretty large files from an s3 bucket, How do we know If the download is completed?
If they aren't we have to take action to delete them, because when logstash start again it will read the file and it will be incomplete and the sincedb will be updated it will never redownload the files again possibly losing events.

If you look at `download_remote_file` or `process_local_log` the changes arent that big.
Its just a way to know that something went wrong and to get out of the processing loop as quick as I can without throwing any errors.

There is a few unit test covering this part and minimal integration test.